mailbox/event/event-memory/src/main/java/org/apache/james/mailbox/events/delivery/InVmEventDelivery.java

```diff
@@ -83,19 +83,21 @@ private ExecutionStages toExecutionStages(MailboxListener.ExecutionMode executio
     }
 
     private void doDeliverToListener(MailboxListener mailboxListener, Event event) {
-        TimeMetric timer = metricFactory.timer(timerName(mailboxListener));
-        try (Closeable mdc = MDCBuilder.create()
+        if (mailboxListener.isUsed(event)) {
+            TimeMetric timer = metricFactory.timer(timerName(mailboxListener));
```

Comment:
This start being too much indented. Maybe this could be solved by method extraction?